src/foam/nanos/auth/SystemAuthService.js

```diff
@@ -11,7 +11,19 @@ foam.CLASS({
 
   javaImports: [
     'foam.nanos.auth.Group',
-    'javax.security.auth.AuthPermission'
+    'javax.security.auth.AuthPermission',
+    'java.util.Map',
+    'java.util.concurrent.ConcurrentHashMap',
+  ],
+
+  properties: [
+    {
+      class: 'Map',
```

Comment:
I think you could specify the javaType as 'Map<String, Boolean>' to avoid having to cast below.